MapboxCoreNavigation/RouteController.swift

```diff
@@ -472,3 +618,134 @@ extension RouteController: CLLocationManagerDelegate {
+    }
+    
+    func sendCancelEvent() {
+        let eventName = "navigation.cancel"
+
+        var eventDictionary = events.addDefaultEvents(routeController: self)
+        eventDictionary["event"] = eventName
+        eventDictionary["arrivalTimestamp"] = sessionState.arrivalTimestamp?.ISO8601 ?? NSNull()
+
+        events.enqueueEvent(withName: eventName, attributes: eventDictionary)
+        events.flush()
+    }
+    
+    func enqueueFeedbackEvent(type: FeedbackType, description: String?) {
+        let eventName = "navigation.feedback"
+        
+        var eventDictionary = events.addDefaultEvents(routeController: self)
+        eventDictionary["event"] = eventName
+        
+        eventDictionary["userId"] = UIDevice.current.identifierForVendor?.uuidString
+        eventDictionary["feedbackType"] = type.rawValue
+        eventDictionary["description"] = description
+        for (key, value) in routeProgress.currentLegProgress.upcomingManeuverDictionary {
+            eventDictionary["step\(key.uppercaseFirst)"] = value
+        }
+        // FIXME: Replace the above enumeration with the below once https://github.com/mapbox/api-events/issues/265 is fixed
+        // eventDictionary["upcomingStep"] = routeProgress.currentLegProgress.upcomingManeuverDictionary
+        eventDictionary["screenshot"] = captureScreen(scaledToFit: 250)?.base64EncodedString()
+
+        outstandingFeedbackEvents.append(FeedbackEvent(timestamp: Date(), eventDictionary: eventDictionary))
+    }
+    
+    func enqueueRerouteEvent() {
+        let eventName = "navigation.reroute"
+
+        let timestamp = Date()
+        
+        var eventDictionary = events.addDefaultEvents(routeController: self)
+        eventDictionary["event"] = eventName
+        
+        eventDictionary["secondsSinceLastReroute"] = sessionState.lastReroute != nil ? round(timestamp.timeIntervalSince(sessionState.lastReroute!)) : -1
+        for (key, value) in routeProgress.currentLegProgress.upcomingManeuverDictionary {
+            eventDictionary["step\(key.uppercaseFirst)"] = value
+        }
+        // FIXME: Replace the above enumeration with the below once https://github.com/mapbox/api-events/issues/265 is fixed
+        // eventDictionary["upcomingStep"] = routeProgress.currentLegProgress.upcomingManeuverDictionary
+        
+        // These are placeholders until the RouteProgress is updated after rerouting
+        eventDictionary["newDistanceRemaining"] = -1
+        eventDictionary["newDurationRemaining"] = -1
+        eventDictionary["newGeometry"] = nil
+        eventDictionary["screenshot"] = captureScreen(scaledToFit: 250)?.base64EncodedString()
+        
+        sessionState.lastReroute = timestamp
+        sessionState.numberOfReroutes += 1
+        
+        outstandingFeedbackEvents.append(RerouteEvent(timestamp: timestamp, eventDictionary: eventDictionary))
+    }
+    
+    func checkAndSendOutstandingFeedbackEvents(forceAll: Bool = false) {
```

Comment:
It isn’t necessary to provide a default value here. Make all the call sites explicitly say whether to force all or not.